io/pci/dlpar.py

```diff
@@ -0,0 +1,293 @@
+
+    def __str__(self):
+        return "Command '%s' exited with %d.\nOutput:\n%s" \
+               % (self.command, self.exitcode, self.output)
+
+
+class DlparPci(Test):
+    '''
+    DLPAR PCI script does pci add,remove and also move operation from one
+    lpar to another lpar. Update the details in yaml file.
+    For move operation, please configure another lpar and update in yaml file.
+    And also make sure both rsct and rsct_rm services up and running
+    '''
+
+    def setUp(self):
+        '''
+        set up required packages and gather necessary test inputs
+        '''
+        sm = SoftwareManager()
+        detected_distro = distro.detect()
+        self.log.info("Test is running on:" + detected_distro.name)
+        if not sm.check_installed("ksh") and not sm.install("ksh"):
+            self.error('ksh is needed for the test to be run')
+        if detected_distro.name == "Ubuntu":
+            if not sm.check_installed("python-paramiko") and not \
+                                      sm.install("python-paramiko"):
+                self.error('python-paramiko is needed for the test to be run')
+            ubuntu_url = self.params.get('ubuntu_url', default=None)
+            debs = self.params.get('debs', default=None)
+            for deb in debs:
+                deb_url = os.path.join(ubuntu_url, deb)
+                deb_install = self.fetch_asset(deb_url, expire='7d')
+                shutil.copy(deb_install, self.srcdir)
+                process.system("dpkg -i %s/%s" % (self.srcdir, deb),
+                               ignore_status=True, sudo=True)
+        else:
+            url = 'http://ausgsa.ibm.com/projects/p/poweryum/ibmtools/ibmtools'
+            url = self.params.get('url', default=url)
+            rpm_install = self.fetch_asset(url, expire='7d')
+            shutil.copy(rpm_install, self.srcdir)
+            os.chdir(self.srcdir)
+            process.run('chmod +x ibmtools')
+            process.run('./ibmtools --install --managed')
+        self.hmc_ip = self.params.get("hmc_ip", '*', default=None)
+        self.hmc_pwd = self.params.get("hmc_pwd", '*', default=None)
+        self.hmc_username = self.params.get("hmc_username", '*', default=None)
+        self.lpar_1 = self.params.get("lpar_1", '*', default=None)
+        self.lpar_2 = self.params.get("lpar_2", '*', default=None)
+        self.pci_device = self.params.get("pci_device", '*', default=None)
+        self.server = self.params.get("server", '*', default=None)
+        self.loc_code = pci.get_slot_from_sysfs(self.pci_device)
+        if self.loc_code is None:
+            self.cancel("Failed to get the location code for the pci device")
+        self.login(self.hmc_ip, self.hmc_username, self.hmc_pwd)
+        self.run_command("uname -a")
+        cmd = 'lshwres -r io -m ' + self.server + \
+              ' --rsubtype slot --filter lpar_names=' + self.lpar_1 + \
+              ' -F drc_index,lpar_id,drc_name | grep -i %s ' % self.loc_code
```

Comment:
the above cmd can also be written like this
`cmd = 'lshwres -r io -m %s --rsubtype slot --filter lpar_names=%s  -F drc_index,lpar_id,drc_name | grep -i %s'  % (self.server, self.lpar_1, self.loc_code)`

Reply:
@abdhaleegit any difference does it make??

Reply:
No It does not make any difference : but more of clean formatting of strings :-)